src/panels/config/dashboard/ha-config-dashboard.ts

```diff
@@ -18,30 +19,35 @@ import {
   TemplateResult,
 } from "lit";
 import { customElement, property, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
 import { isComponentLoaded } from "../../../common/config/is_component_loaded";
+import { computeStateDomain } from "../../../common/entity/compute_state_domain";
+import "../../../components/ha-button-menu";
 import "../../../components/ha-card";
-import "../../../components/ha-icon-next";
 import "../../../components/ha-icon-button";
+import "../../../components/ha-icon-next";
 import "../../../components/ha-menu-button";
-import "../../../components/ha-button-menu";
 import "../../../components/ha-svg-icon";
 import { CloudStatus } from "../../../data/cloud";
-import {
-  refreshSupervisorAvailableUpdates,
-  SupervisorAvailableUpdates,
-} from "../../../data/supervisor/root";
+import { updateCanInstall, UpdateEntity } from "../../../data/update";
+import { showAlertDialog } from "../../../dialogs/generic/show-dialog-box";
 import { showQuickBar } from "../../../dialogs/quick-bar/show-dialog-quick-bar";
 import "../../../layouts/ha-app-layout";
 import { haStyle } from "../../../resources/styles";
 import { HomeAssistant } from "../../../types";
+import { documentationUrl } from "../../../util/documentation-url";
+import { showToast } from "../../../util/toast";
 import "../ha-config-section";
 import { configSections } from "../ha-panel-config";
 import "./ha-config-navigation";
 import "./ha-config-updates";
-import { fireEvent } from "../../../common/dom/fire_event";
-import { showAlertDialog } from "../../../dialogs/generic/show-dialog-box";
-import { showToast } from "../../../util/toast";
-import { documentationUrl } from "../../../util/documentation-url";
+
+const updateEntities = memoizeOne(
```

Comment:
Put the memoized functions on the class instances such that running this class doesn't change any module variables.

Reply:
We might not run the dashboard twice, but people might copy the pattern from here.

Reply:
Also it's _slightly_ faster when the dashboard is not shown, as memoizeOne is only executed when class is instantiated if set as instance var.